### PR TITLE
DEVX-2191: fix removal workflow

### PR DESCRIPTION
**Pull request by AmyJZhao**
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/remove-template.yml [hunk 1]
```diff
@@ -67,7 +67,7 @@ jobs:
     outputs:
       error: ${{ steps.run-remove-from-registry.outputs.error }}
     needs: [parse-issue-form-body, validate-requestor, validate-admin]
-    if: ${{ always() && (needs.validate-requestor.outputs.error == '' || needs.validate-admin.outputs.is-admin == 'active') }}
+    if: ${{ always() && contains(join(needs.parse-issue-form-body.result, ','), 'success') && (needs.validate-requestor.outputs.error == '' || needs.validate-admin.outputs.is-admin == 'active') }}
```
Comment (slitviachenko) [inline .github/workflows/remove-template.yml:70]: @AmyJZhao , you don't need to join `needs.parse-issue-form-body.result` as it is always `string`, you can use
```
needs.parse-issue-form-body.result == 'success'
```

File: .github/workflows/remove-template.yml [hunk 2]
```diff
@@ -96,8 +96,8 @@ jobs:
   fail-removal-comment:
     name: Removal failed
     runs-on: ubuntu-latest
-    needs: [validate-requestor, remove-from-registry]
-    if: ${{ always() && (needs.remove-from-registry.outputs.error != '' || needs.validate-requestor.outputs.error != '') }}
+    needs: [validate-requestor, validate-admin, remove-from-registry]
+    if: ${{ always() && (needs.remove-from-registry.outputs.error != '' || (needs.validate-requestor.outputs.error != '' && needs.validate-admin.outputs.is-admin != 'active')) }}
```
Comment (slitviachenko) [inline .github/workflows/remove-template.yml:100]: @AmyJZhao , it looks like "validate-requestor" should depend on "validate-admin".
Does it make sense to run 2 these checks in parallel?

**Review by slitviachenko — CHANGES_REQUESTED** [review]

File: .github/workflows/remove-template.yml (hunk 1 above)
Comment (AmyJZhao) [reply .github/workflows/remove-template.yml]: Done

File: .github/workflows/remove-template.yml (hunk 2 above)
Comment (AmyJZhao) [reply .github/workflows/remove-template.yml]: I updated the workflow to do validate-admin (passed) -> removal & validate-admin (failed) -> validate-requestor (passed) -> removal

**Comment by AmyJZhao** [comment]
Test:

Removal success (am admin): https://github.com/AmyJZhao/aio-template-submission/actions/runs/2116372425

**Review by slitviachenko — APPROVED** [review]

**Review by lenaorobei — APPROVED** [review]

**Pull request merged**